sdk/src/main/scala/com/horizen/consensus/NonceConsensusEpochInfo.scala

```diff
@@ -29,5 +29,5 @@ case class NonceConsensusEpochInfo(consensusNonce: ConsensusNonce) extends Bytes
 object NonceConsensusEpochInfoSerializer extends ScorexSerializer[NonceConsensusEpochInfo]{
   override def serialize(obj: NonceConsensusEpochInfo, w: Writer): Unit = w.putBytes(obj.consensusNonce)
 
-  override def parse(r: Reader): NonceConsensusEpochInfo = NonceConsensusEpochInfo(ConsensusNonce @@ r.getBytes(consensusNonceLength))
+  override def parse(r: Reader): NonceConsensusEpochInfo = NonceConsensusEpochInfo(ConsensusNonce @@ r.getBytes(r.remaining))
```

Comment:
Why don't we get fixed size of consensusNonce based on consensus epoch?

Reply:
There's not enough context in this place to determine current consensus epoch

Reply:
Also let me share my comment from discord:

> The reason for that is that nonce length is going to change after the fork(8 -> 32), and we have too many places where this class is used. To get current epoch and thus actual consensus nonce length, we'll have to add code to determine current epoch and fork to all of them. It's not looking good and greatly complicates the code. Also I can't also make this change in one place, cause around the fork node may encounter both new and old nonce values.

Reply:
Double checked usage of NonceConsensusEpochInfoSerializer.

Yes, we can keep this implementation.